bigquery/google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -72,6 +54,46 @@ def scalar_to_query_parameter(value, name=None):
     return bigquery.ScalarQueryParameter(name, parameter_type, value)
 
 
+def array_to_query_parameter(value, name=None):
+    """Convert an array-like value into a query parameter.
+
+    Args:
+        value (Sequence[Any]): The elements of the array (should not be a
+            string-like Sequence).
+        name (Optional[str]): Name of the query parameter.
+
+    Returns:
+        A query parameter corresponding with the type and value of the plain
+        Python object.
+
+    Raises:
+        :class:`~google.cloud.bigquery.dbapi.exceptions.ProgrammingError`
+        if the type of array elements cannot be determined.
+    """
+    if not array_like(value):
+        raise exceptions.ProgrammingError(
+            "The value of parameter {} must be a sequence that is "
+            "not string-like.".format(name)
+        )
+
+    if not value:
+        raise exceptions.ProgrammingError(
+            "Encountered an empty array-like value of parameter {}, cannot "
```

Comment:
This concerns me, since someone is probably going to hit this pretty quickly, but I don't like the alternatives (such as defaulting to STRING parameters or something).

Reply:
I hope the error message is clear enough?
I don't like an implicit fallback to strings either, but also can't think of a better way of handling this right now (in the helper itself at least).